securedrop/tests/functional/test_source_session_timeout.py

```diff
@@ -1,17 +1,56 @@
-from . import source_navigation_steps
-from . import functional_test
+import time
+from pathlib import Path
 
+import pytest
 
-class TestSourceSessions(
-        functional_test.FunctionalTest,
-        source_navigation_steps.SourceNavigationStepsMixin):
+from tests.functional.conftest import spawn_sd_servers
+from tests.functional.app_navigators import SourceAppNagivator
+from tests.functional.factories import SecureDropConfigFactory
 
-    session_expiration = 5
 
-    def test_source_session_timeout(self):
-        self._source_visits_source_homepage()
-        self._source_clicks_submit_documents_on_homepage()
-        self._source_continues_to_submit_page()
-        self._source_waits_for_session_to_timeout()
-        self.driver.refresh()
-        self._source_sees_session_timeout_message()
+# Very short session expiration time
+SESSION_EXPIRATION_SECONDS = 3
+
+
+@pytest.fixture(scope="session")
+def sd_servers_with_short_session_timeout(setup_journalist_key_and_gpg_folder):
+    """Spawn the source and journalist apps as separate processes with a short session timeout."""
+    # Generate a securedrop config with a very short session timeout
+    config_with_short_timeout = SecureDropConfigFactory.create(
+        SESSION_EXPIRATION_MINUTES=SESSION_EXPIRATION_SECONDS / 60,
+        SECUREDROP_DATA_ROOT=Path("/tmp/sd-tests/functional-session-timeout"),
+    )
+
+    # Ensure the GPG settings match the one in the config to use, to ensure consistency
+    journalist_key_fingerprint, gpg_dir = setup_journalist_key_and_gpg_folder
+    assert Path(config_with_short_timeout.GPG_KEY_DIR) == gpg_dir
+    assert config_with_short_timeout.JOURNALIST_KEY == journalist_key_fingerprint
+
+    # Spawn the apps in separate processes
+    with spawn_sd_servers(config_to_use=config_with_short_timeout) as sd_servers_result:
+        yield sd_servers_result
```

Comment:
The above is the "killer" feature of the new fixture and factory code: the source and journalist apps to be spawned will use the supplied `config_to_use` regardless of the content of `securedrop/config.py`. 

The confusion that comes from having to deal with live SDConfig objects that don't match the content of `securedrop/config.py` has been one of the biggest challenges for me when updating the test suite. This also comes from having different parts of the code sometimes use the live object or sometimes `securedrop/config.py`. 

I think one medium term goal (started with some of the changes here) is to get rid of `securedrop/config.py` in the context of the test suite.